File: public_encapsulation.py
class Animal:
    legs=None
    type=None
    
    def __init__(self, legs, type) -> None:
        self.legs= legs
        self.type= type
        
    def __str__(self) -> str:
        return f"{self.type} type of animal has {self.legs} legs"
    
    #python treats this as a regular function instead of class method
    def display(name):
        print(name)
        
cow= Animal(4, "herbivore")
cow.display() #this passes the name of the obj as parameter

File: public_encapsulation.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;

class Animal{
    public:
    int legs;
    string type;

    Animal(int legs, string type){
        this->legs=legs;
        this->type= type;
    }

    void display(string name){
        cout<<name<<" animal has "<<legs<<" legs and is a "<<type<<" type"<<endl;
    }

};

int main(){

    Animal cow= Animal(4,"herbivore");
    cow.display("cow");

   
return 0;
}